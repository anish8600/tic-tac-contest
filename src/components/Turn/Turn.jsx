import React from 'react';
import { TIC_TAC_CONSTANTS, MAX_ATTEMPTS } from '../../helper/tictacHelper';
import PropTypes from 'prop-types';

const Turn = (props) => {
    let turnIndicator;
    if (props.champion) {
      turnIndicator = TIC_TAC_CONSTANTS.WINNER_MSG_1 + props.champion + TIC_TAC_CONSTANTS.WINNER_MSG_2;
    } else if(props.steps < MAX_ATTEMPTS){
      turnIndicator = TIC_TAC_CONSTANTS.TURN_MSG + (props.turn ? TIC_TAC_CONSTANTS.X_SYMBOL : TIC_TAC_CONSTANTS.O_SYMBOL);
    } else{
      turnIndicator = TIC_TAC_CONSTANTS.TIE_MSG;
    }
    return <h2>{turnIndicator}</h2>;
}

Turn.propTypes = { champion: PropTypes.string.isRequired, 
                   turn: PropTypes.string.isRequired,
                   steps: PropTypes.number.isRequired
                 };

export default Turn;