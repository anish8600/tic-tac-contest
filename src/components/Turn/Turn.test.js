import React from 'react';
import { shallow } from 'enzyme';
import Turn from './Turn';

describe('Turn', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Turn champion={'X'} turn={'X'} steps={3}/>));

  it('should render a <h2 />', () => {
    expect(wrapper.find('h2').length).toEqual(1);
  });

  it('should display correct message for winner>', () => {
    expect(wrapper.find('h2').text()).toEqual('Congratulations X.You are a Winner!!!!!!!!');
  });

  it('should display correct message for winner>', () => {
    wrapper.setProps({ champion:null })
    expect(wrapper.find('h2').text()).toEqual('Your turn: X');
  });

  it('should display correct message for tie scenario>', () => {
    wrapper.setProps({ champion:null, steps:10 })
    expect(wrapper.find('h2').text()).toEqual('There is tie. Click \'Reset Game\' to play again.');
  });
});