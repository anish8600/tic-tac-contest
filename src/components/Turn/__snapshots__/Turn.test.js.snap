// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Turn should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Turn
    champion="X"
    steps={3}
    turn="X"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Congratulations X.You are a Winner!!!!!!!!",
    },
    "ref": null,
    "rendered": "Congratulations X.You are a Winner!!!!!!!!",
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Congratulations X.You are a Winner!!!!!!!!",
      },
      "ref": null,
      "rendered": "Congratulations X.You are a Winner!!!!!!!!",
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
